sdk/react/discoveryPanel.js

```diff
@@ -119,6 +115,32 @@ var DiscoveryPanel = React.createClass({
     return (
       <Animated.View style={[styles.panel, animationStyle]}>
         {this.renderHeader()}
+        {this.renderList()}
+        {this.renderError()}
+      </Animated.View>
+    );
+  },
+
+  _isDicoveryError: function() {
+    return this.props.dataSource == null || this.props.dataSource.length == 0;
+  },
+
+  renderList: function() {
+    var panelHeight = this.props.height - 40;
+    var renderHorizontal = Utils.shouldShowLandscape(this.props.width, this.props.height);
+    if (this._isDicoveryError()) {
+      return (
+        <ResponsiveList
+          horizontal={false}
+          data={null}
```

Comment:
To understand a bit more, why do you use null here? If discovery has an error, it means that props.dataSource is empty, right?